pkg/solver/server.go

```diff
@@ -376,6 +379,145 @@ func (server *solverServer) addJobOffer(jobOffer data.JobOffer, res corehttp.Res
 	return server.controller.addJobOffer(jobOffer)
 }
 
+func (server *solverServer) addJobOfferWithFiles(res corehttp.ResponseWriter, req *corehttp.Request) {
+	// Check signature and validate the job offer
+	signerAddress, err := http.CheckSignature(req)
+	if err != nil {
+		server.log.Error().Err(err).Msg("error checking signature")
+		corehttp.Error(res, "Unauthorized", corehttp.StatusUnauthorized)
+		return
+	}
+
+	// Set a maximum size for the entire request body
+	// This includes the job offer, which only accounts for a small portion of the request body
+	maxRequestSize := int64(server.options.Storage.MaximumFileInputsSizeMB << 20)
+	req.Body = corehttp.MaxBytesReader(res, req.Body, maxRequestSize)
+
+	// Parse the multipart form with max memory size
+	err = req.ParseMultipartForm(int64(server.options.Storage.MaximumFileInputsMemoryMB << 20))
```

Comment:
Looks like you have `int64(server.options.Storage.MaximumFileInputsMemoryMB << 20)` computed already with `maxRequestSize` so you can probs re-use that unless i've missed a detail

Reply:
This one uses the `MaximumFileInputsMemoryMB` config, where `maxRequestSize` uses `MaximumFileInputsSizeMB` (names are quite close, `Memory` vs `Size`).

Reply:
Ahh yeah, there indeed was a detail i missed 😆